DOCUMENTATION.md

```diff
@@ -0,0 +1,305 @@
+
+It is recommended to add the class manually to the `body` element though, because that means that users without JavaScript will also get fallback styles. When impress.js script detects that the browser supports all required features, the `.impress-not-support` class will be removed from the `body` element.
+
+**Example:**
+
+```CSS
+.impress-not-supported .step {
+  display: inline-block;
+}
+```
+
+## JavaScript
+
+### impress().init()
+
+Initializes impress.js globally in the page. Only one instance of impress.js is supported per document.
+
+**Example:**
+
+```JavaScript
+impress().init();
+```
+
+Triggers the `impress:init` event in the [Root Element](#root-element) after the presentation is initialized.
+
+**Example:**
+
+```JavaScript
+var rootElement = document.getElementById( "impress" );
+rootElement.addEventListener( "impress:init", function() {
+  console.log( "Impress init" );
+});
+impress().init();
+```
+
+### impress().next()
+
+Navigates to the next step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.next();
+```
+
+### impress().prev()
+
+Navigates to the previous step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.prev();
+```
+
+### impress().goto(stepIndex|stepElementId|stepElement, [duration])
+
+Accepts a [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) that represents the step index.
+
+Navigates to the step given the provided step index.
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.goto(7);
+```
+
+Accepts a [`String`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) that represents the [Step Element](#step-element) id.
+
+Navigates to the step given the provided [Step Element](#step-element) id.
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.goto( "overview" );
+```
+
+Accepts an [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement) that represents the [Step Element](#step-element).
+
+Navigates to the step given the provided [Step Element](#step-element).
+
+**Example:**
+
+```JavaScript
+var overview = document.getElementById( "overview" );
+var api = impress();
+api.init();
+api.goto( overview );
+```
+
+Accepts an optional [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) in the last argument that represents the duration of the transition in milliseconds. If not provided, the default transition duration for the presentation will be used.
+
+Triggers the `impress:stepenter` event in the [Root Element](#root-element) when the presentation navigates to the target [Step Element](#step-element).
```

Comment:
So, you can add this listener either in the body or in the root element. Nor sure where this should be but it seems to make sense documenting that it should be in the Root Element, because we don't want to unnecessarily add things to the `body`, which makes extension of impress.js pretty hard (you can have only one `body` per page, and that means only one instance per page).